AdGoBye/Updater.cs

```diff
@@ -0,0 +1,132 @@
+using System.Diagnostics.CodeAnalysis;
+using System.Net;
+using System.Reflection;
+using System.Text.Json;
+using Semver;
+using Serilog;
+
+namespace AdGoBye;
+
+public static class Updater
+{
+    private const string ConnectivityCheckUrl = "http://connectivity-check.ubuntu.com/";
+    private const string GithubReleaseUrl = "https://api.github.com/repos/AdGoBye/AdGoBye/releases/latest";
+    private const string ETagFile = "ReleaseETag";
+
+    private static readonly HttpClient Client = CreateHttpClient();
+    private static readonly ILogger Logger = Log.ForContext(typeof(Updater));
+
+    public static void UpdateConditionally()
+    {
+        if (!CheckConnectivity())
+        {
+            Logger.Warning("We appear to be offline, skipping update check.");
+            return;
+        }
+
+        if (GetVersionFromRemote() is not { } remoteVersion) return;
+
+        var remoteVersionSemVer = SemVersion.Parse(remoteVersion.tag_name.Replace("v", ""));
+        var localVersionSemVer = SemVersion.Parse(GetSelfVersion());
+        Logger.Debug("Remote: {remote}, Local: {local} ", remoteVersion.tag_name, localVersionSemVer);
+        if (!localVersionSemVer.IsPrerelease && localVersionSemVer.ComparePrecedenceTo(remoteVersionSemVer) <= 0)
+        {
+            Logger.Information("Version {remoteVersion} is out, you are using {localVersion}, download it at {url}",
```

Comment:
I think something like "New version {remoteVersion} is out...." would work better. I'm used to seeing the word `New` with every update notification.

Reply:
So essentially:
```c#
Logger.Information("New version {remoteVersion} is out, you are using {localVersion}, download it at {url}",
```
is what you're suggesting?

Reply:
Yup, also change `download it at {url}` to maybe like, `download the new version at {}`, just grammatical since `it` can become a bit misleading there.

Reply:
Applied verbatim.